examples/collection/assembly/index.ts

```diff
@@ -0,0 +1,71 @@
+import { inference, collections } from "@hypermode/functions-as";
+
+// This model name should match one defined in the hypermode.json manifest file.
+const modelName: string = "my-custom-embedding";
+const myProducts: string = "myProducts";
+const searchMethods: string[] = ["searchMethod1", "searchMethod2"];
+
+// This function takes input text and returns the vector embedding for that text.
+export function embed(text: string): f64[] {
+  return inference.getTextEmbedding(modelName, text);
+}
+
+export function upsertProduct(description: string): string {
+  const response = collections.upsert(myProducts, null, description);
+  if (response.status !== "success") {
+    return response.error;
+  }
+  return response.key;
+}
+
+export function deleteProduct(key: string): string {
+  const response = collections.remove(myProducts, key);
+  if (response.status !== "success") {
+    return response.error;
+  }
+  return response.status;
```

Comment:
There's a few things problematic here.

- The user would have to know to check for that specific string, `"success"` exactly lower case, etc.  There's nothing to guide them to what the correct values are.
- Errors and success strings are returned identically from this example, so there's they always will look like a successful function run in the logs and in the result.

It should instead be something like:

```suggestion
  if (response.isSuccessful) {
    throw new Error(response.error);
  }
  
  return response.status;
```

There's actually no need to return anything at all, but the final return will at least give some confidence value in the resulting GraphQL call.

Reply:
The above applies throughout the examples.  I'll leave some additional feedback for the implementation.